### PR TITLE
Add symbol listener activity

### PR DESCRIPTION
**Pull request by samfader**
Resolves https://github.com/mapbox/mapbox-android-demo/issues/995

_Langston edit: added the GIF_
ezgif com-resize

### CODE REVIEW

**Comment by langsmith** [comment]
Congrats. In the flow of something else, so will probably review tomorrow

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_annotation_plugin_symbol_activity);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: nit: delete this blank line 😬

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+              String.format("Symbol long clicked"),
+              Toast.LENGTH_SHORT).show();
+            symbol.setIconImage(MAKI_ICON_AIRPORT);
+            symbolManager.update(symbol);
+          }
+        }));
+
+        symbolManager.addDragListener(new OnSymbolDragListener() {
+          @Override
+          public void onAnnotationDragStarted(Symbol annotation) {
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: Leave a comment saying that `onAnnotationDragStarted()` is left empty on purpose

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+          }
+
+          @Override
+          public void onAnnotationDrag(Symbol symbol) {
+          }
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: Leave a comment saying that `onAnnotationDrag()` is left empty on purpose

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+          @Override
+          public void onAnnotationDragFinished(Symbol annotation) {
+          }
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: Leave a comment saying that `onAnnotationDragFinished()` is left empty on purpose

File: MapboxAndroidDemo/src/main/res/values/descriptions_strings.xml [hunk 5]
```diff
@@ -81,6 +81,7 @@
     <string name="activity_plugins_building_plugin_description">Use the building plugin to easily display 3D building height</string>
     <string name="activity_plugins_geojson_plugin_description">Easily retrieve GeoJSON data from a url, asset, or path</string>
     <string name="activity_plugins_places_plugin_description">Add location search ("geocoding") functionality and UI to search for any place in the world</string>
+    <string name="activity_plugins_symbol_listener_description">Listen for Symbol interaction using the Annotations plugin and built-in listeners</string>
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/res/values/descriptions_strings.xml]:
```suggestion
    <string name="activity_plugins_symbol_listener_description">Listen for Symbol interaction using the Annotation plugin and its built-in listeners</string>
```

File: MapboxAndroidDemo/build.gradle [hunk 6]
```diff
@@ -10,6 +10,11 @@ android {
     compileSdkVersion androidVersions.compileSdkVersion
     buildToolsVersion androidVersions.buildToolsVersion
 
+    compileOptions {
```
Comment (langsmith) [inline MapboxAndroidDemo/build.gradle]: I don't think you need this `compileOptions`section anymore. I tried removing it and the example works just fine for me.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private static final String LOG = "SymbolListenerActivity";
+  private static final String MAKI_ICON_CAFE = "cafe-15";
+  private static final String MAKI_ICON_HARBOR = "harbor-15";
+  private static final String MAKI_ICON_AIRPORT = "airport-15";
+  private SymbolManager symbolManager;
+  private Symbol symbol;
+  private final List<ValueAnimator> animators = new ArrayList<>();
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: You can delete this. You're not using it anywhere in the example.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+import java.util.List;
+
+/**
+ * Change symbol icon by pressing on icon
+ */
+public class SymbolListenerActivity extends AppCompatActivity implements
+  OnMapReadyCallback {
+
+  private MapView mapView;
+  private MapboxMap mapboxMap;
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: You can delete this. Field can be converted to a local variable (on line 62).

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+
+  }
+
+  @Override
+  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+
+    this.mapboxMap = mapboxMap;
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/SymbolListenerActivity.java]: You can delete this. You don't use `mapboxMap` elsewhere.

**Comment by samfader** [comment]
Thanks @langsmith! Let me know what you think now.

**Comment by samfader** [comment]
@langsmith It looks like Circle is failing and I think it's due to the compileOptions code that was removed here: https://github.com/mapbox/mapbox-android-demo/pull/1000#pullrequestreview-219577095

**Comment by langsmith** [comment]
Ok, then let's add that `compileOptions` thing back in @samfader

**Comment by samfader** [comment]
@langsmith Passing now. OK to merge?

**Comment by langsmith** [comment]
:shipit:

**Pull request merged**